scheduler/app/libs/bitquery.py

```diff
@@ -48,19 +50,31 @@ async def get_eth_transactions(address: str) -> list[BitqueryTransfer]:
     else:
 
         async def get_balance_hist_data():
-            async with AsyncClient(headers={"X-API-KEY": BITQUERY_API_KEY}) as client:
+            timeout = Timeout(10.0, read=15.0, connect=30.0)
+            async with AsyncClient(
+              headers={"X-API-KEY": BITQUERY_API_KEY},
+              timeout=timeout
+            ) as client:
                 resp = await client.post(
                     BITQUERY_URL,
-                    json={"query": ETH_QUERY_TEMPLATE.replace("$address", address)},
+                    json={
+                      "query": ETH_QUERY_TEMPLATE.replace("$address", address)
+                    },
                 )
                 resp.raise_for_status()
-                return resp.json()["data"]["ethereum"]["address"][0]["balances"][0][
-                    "history"
-                ]
+                try:
+                    data = resp.json()["data"]
+                    return data["ethereum"]["address"][0]["balances"][0][
+                      "history"
+                    ]
+                except TypeError as e:
+                    print_exception(type(e), e, e.__traceback__)
+                    return None
 
         balance_hist_data = await get_balance_hist_data()
         db.hset(CACHE_HASH, cache_key, json.dumps(balance_hist_data))
-
+    if not balance_hist_data:
```

Comment:
this if statement supresses an error that would otherwise be raised in this return statement
```python
return [
        BitqueryTransfer(
            dateutil.parser.parse(hist_item["timestamp"]),
            hist_item["transferAmount"],
            hist_item["value"],
        )
        for hist_item in balance_hist_data
    ]
```
if `balance_hist_data is None` then it raises.

If anything, I could've expressed this and the `if` statement better. However if you have a better fix in mind I'm open to suggestions 🙂.

Reply:
Sorry wrote this comment too fast 😅

I think I had in mind that we want to return `[]`. That allows keeping the data structures unchanged, and thus not changing `populate_bitquery_hist_eth_balance`, isn't it?

Reply:
I think that's best.